rust/processor/src/processors/econia_processor.rs

```diff
@@ -0,0 +1,200 @@
+            .on_conflict_do_nothing(),
+        None,
+    )?;
+    Ok(())
+}
+
+#[async_trait]
+impl ProcessorTrait for EconiaTransactionProcessor {
+    fn name(&self) -> &'static str {
+        NAME
+    }
+
+    async fn process_transactions(
+        &self,
+        transactions: Vec<Transaction>,
+        start_version: u64,
+        end_version: u64,
+        _: Option<u64>,
+    ) -> anyhow::Result<ProcessingResult> {
+        let mut conn = self.get_conn();
+        let (_, txn_details, _, _, _) = TransactionModel::from_transactions(&transactions);
+
+        // Create a hashmap to store block_height to timestamp.
+        let mut block_height_to_timestamp: HashMap<i64, DateTime<Utc>> = HashMap::new();
+
+        // Iterate through the transactions and populate the map.
+        for txn_detail in txn_details {
+            match txn_detail {
+                TransactionDetail::User(user_transaction, _) => {
+                    block_height_to_timestamp.insert(
+                        user_transaction.block_height,
+                        DateTime::from_utc(user_transaction.timestamp, Utc),
+                    );
+                },
+                TransactionDetail::BlockMetadata(block_metadata_transaction) => {
+                    block_height_to_timestamp.insert(
+                        block_metadata_transaction.block_height,
+                        DateTime::from_utc(block_metadata_transaction.timestamp, Utc),
+                    );
+                },
+            }
+        }
+
+        let market_registration_type =
+            format!("{}::registry::MarketRegistrationEvent", self.econia_address);
+        let mut market_registration_events = vec![];
+        for transaction in transactions {
+            let (txn, _, events, _, _) = TransactionModel::from_transaction(&transaction);
+            for (index, event) in events.iter().enumerate() {
+                if event.type_ != market_registration_type {
+                    continue;
+                }
+                let txn_version = BigDecimal::from(txn.version);
+                let event_idx = BigDecimal::from(index as u64);
+                let market_id = opt_value_to_big_decimal(event.data.get("market_id"));
+                let lot_size = opt_value_to_big_decimal(event.data.get("lot_size"));
+                let tick_size = opt_value_to_big_decimal(event.data.get("tick_size"));
+                let min_size = opt_value_to_big_decimal(event.data.get("min_size"));
+                let underwriter_id = opt_value_to_big_decimal(event.data.get("underwriter_id"));
+                let time = *block_height_to_timestamp
+                    .get(&event.transaction_block_height)
+                    .unwrap();
+                let base_name_generic;
+                let base_account_address;
+                let base_module_name;
+                let base_struct_name;
+                if value_is_not_empty_string(&event.data["base_name_generic"]) {
```

Comment:
I find this check confusing

Maybe something like if_has_string?

Reply:
changed to value_is_empty_string and switched the branches